generator_process.py

```diff
@@ -0,0 +1,150 @@
+import json
+import subprocess
+import sys
+import os
+import numpy as np
+
+class GeneratorProcess():
+    def __init__(self):
+        self.process = subprocess.Popen([sys.executable,'generator_process.py'],cwd=os.path.dirname(os.path.realpath(__file__)),stdin=subprocess.PIPE,stdout=subprocess.PIPE)
+    
+    def kill(self):
+        self.process.kill()
+    
+    def prompt2image(
+            self,
+            prompt,
+            iterations       = None,
+            steps            = None,
+            seed             = None,
+            cfg_scale        = None,
+            ddim_eta         = None,
+            width            = None,
+            height           = None,
+            sampler_name     = None,
+            seamless         = False,
+            init_img         = None,
+            fit              = False,
+            strength         = None,
+            full_precision   = False,
+            step_callback    = None,
+            image_callback   = None):
+        b = bytes(json.dumps({
```

Comment:
Is it possible to use something like `vars(dream_prompt)` to get the full prompt metadata as a dict instead of passing all the parameters each time? Then the `dream_prompt.py` model can be updated in one place.

Reply:
vars() and dict() both come out empty unfortunately
![image](https://user-images.githubusercontent.com/47096043/192069962-e947cdc5-2b66-496d-a013-6bd493f92b13.png)

Reply:
found a way that's still decently short
![image](https://user-images.githubusercontent.com/47096043/192070865-b49dd27d-0332-4f2a-a790-50a421bc064f.png)
just need to get keys from `dream_prompt.py attributes` instead of an inline list

Reply:
Does `dream_prompt.__annotations__` do it?

Reply:
Looks like `bpy.context.scene.dream_textures_prompt.__annotations__.keys()` may do the trick:
```python
dict_keys(['prompt_structure', 'width', 'height', 'seamless', 'show_advanced', 'random_seed', 'seed', 'full_precision', 'iterations', 'steps', 'cfgscale', 'sampler', 'show_steps', 'use_init_img', 'use_inpainting', 'strength', 'fit', 'prompt_structure_token_subject', 'prompt_structure_token_subject_enum', 'prompt_structure_token_framing', 'prompt_structure_token_framing_enum', 'prompt_structure_token_position', 'prompt_structure_token_position_enum', 'prompt_structure_token_film_type', 'prompt_structure_token_film_type_enum', 'prompt_structure_token_camera_settings', 'prompt_structure_token_camera_settings_enum', 'prompt_structure_token_shooting_context', 'prompt_structure_token_shooting_context_enum', 'prompt_structure_token_lighting', 'prompt_structure_token_lighting_enum', 'prompt_structure_token_subject_type', 'prompt_structure_token_subject_type_enum', 'prompt_structure_token_genre', 'prompt_structure_token_genre_enum'])
```

Reply:
`{'prompt_structure': 'texture', 'width': 512, 'height': 512, 'seamless': False, 'show_advanced': False, 'random_seed': True, 'seed': '0', 'full_precision': False, 'iterations': 1, 'steps': 25, 'cfgscale': 7.5, 'sampler': 'k_lms', 'show_steps': True, 'use_init_img': False, 'use_inpainting': False, 'strength': 0.75, 'fit': True, 'prompt_structure_token_subject': '', 'prompt_structure_token_subject_enum': 
'custom', 'prompt_structure_token_framing': '', 'prompt_structure_token_framing_enum': 'ecu', 'prompt_structure_token_position': '', 'prompt_structure_token_position_enum': 'overhead', 'prompt_structure_token_film_type': '', 'prompt_structure_token_film_type_enum': 'bw', 'prompt_structure_token_camera_settings': '', 'prompt_structure_token_camera_settings_enum': 'high_speed', 'prompt_structure_token_shooting_context': '', 'prompt_structure_token_shooting_context_enum': 'film_still', 'prompt_structure_token_lighting': '', 'prompt_structure_token_lighting_enum': 'golden_hour', 'prompt_structure_token_subject_type': '', 'prompt_structure_token_subject_type_enum': 'environment', 'prompt_structure_token_genre': '', 'prompt_structure_token_genre_enum': 'scifi', 'prompt_structure_token_concept_artist': '', 'prompt_structure_token_concept_artist_enum': 'blevins'}`

a little cluttered but usable